pkg/controllers/user/nsserviceaccount/nssvcaccnt.go

```diff
@@ -0,0 +1,87 @@
+package nsserviceaccount
+
+import (
+	"context"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type namespaceSvcAccountHandler struct {
+	clusterCtx *config.UserContext
```

Comment:
The whole context should not be stored, update to use ServiceAccountLister

Reply:
And a client for the update call

Reply:
LIster does not work for the v1.ServiceAccount object _ i tried using it earlier

Reply:
Specifically the update does not work without using the 
clientv1.ServiceAccountInterface svcAccounts = clusterCtx.K8sClient.CoreV1().ServiceAccounts(namespace)

Also notice that since we need to use this v1 interface, to get the  default serviceaccount of a particular namespace by name, we have to pass in the "namespace" and create a new instance of this interface each time so that the following Get call works.

defSvcAccnt, err := svcAccounts.Get(context.TODO(), "default", metav1.GetOptions{})

Reply:
Why doesn't a lister work?

Reply:
@dramich I tested again changing to lister and types interface and was able to see the updates go through! Not sure why it did not work earlier - I think possibly I was using the userOnlyContext during that test. But now updating to use rancher types instead of client-go K8sClient.CoreV1()